File: good-vs-evil/goodVsEvil.js
const goodVsEvil = (good, evil) => {

    let goodArr = good.split(' ');
    let evilArr = evil.split(' ');

    goodArr = goodArr.map((element, index) => {
        if(index === 0) return Number(element)
        if(index === 1) return Number(element) * 2;
        if(index === 2) return Number(element) * 3;
        if(index === 3) return Number(element) * 3;
        if(index === 4) return Number(element) * 4;
        if(index === 5) return Number(element) * 10;
    });

    evilArr = evilArr.map((element, index) => {
        if(index === 0) return Number(element)
        if(index === 1) return Number(element) * 2;
        if(index === 2) return Number(element) * 2;
        if(index === 3) return Number(element) * 2;
        if(index === 4) return Number(element) * 3;
        if(index === 5) return Number(element) * 5;
        if(index === 6) return Number(element) * 10;
    });

    const goodNum = goodArr.reduce((acc, curr) => 
        acc + curr, 0,
    );

    const evilNum = evilArr.reduce((acc, curr) => 
    acc + curr, 0,
    );

    if(goodNum > evilNum) {
        return 'Battle Result: Good triumphs over Evil';
    } else {
        if(goodNum < evilNum) {
            return 'Battle Result: Evil eradicates all trace of Good';
        } else {
            return 'Battle Result: No victor on this battle field';
        };
    };
};

module.exports = { goodVsEvil };

//https://www.codewars.com/kata/52761ee4cffbc69732000738/train/javascript

File: python-venv/goodVsEvil/goodVsEvil.py
def goodVsEvil(good, evil):

    goodArr = good.split(' ')
    evilArr = evil.split(' ')

    for i in range(len(goodArr)):
        if i == 0:
            goodArr[i] = int(goodArr[i])
        if i == 1:
            goodArr[i] = int(goodArr[i]) * 2
        if i == 2:
            goodArr[i] = int(goodArr[i]) * 3
        if i == 3:
            goodArr[i] = int(goodArr[i]) * 3
        if i == 4:
            goodArr[i] = int(goodArr[i]) * 4
        if i == 5:
            goodArr[i] = int(goodArr[i]) * 10

    for i in range(len(evilArr)):
        if i == 0:
            evilArr[i] = int(evilArr[i])
        if i == 1:
            evilArr[i] = int(evilArr[i]) * 2
        if i == 2:
            evilArr[i] = int(evilArr[i]) * 2
        if i == 3:
            evilArr[i] = int(evilArr[i]) * 2
        if i == 4:
            evilArr[i] = int(evilArr[i]) * 3
        if i == 5:
            evilArr[i] = int(evilArr[i]) * 5
        if i == 6:
            evilArr[i] = int(evilArr[i]) * 10

    if sum(goodArr) > sum(evilArr):
        return "Battle Result: Good triumphs over Evil"
    elif sum(goodArr) < sum(evilArr):
        return "Battle Result: Evil eradicates all trace of Good"
    else:
        return "Battle Result: No victor on this battle field"